src/api_manager/auth/lib/auth_jwt_validator.cc

```diff
@@ -612,7 +652,36 @@ bool JwtValidatorImpl::ExtractPubkeyFromJwk(const grpc_json *jkey) {
   return true;
 }
 
-grpc_jwt_verifier_status JwtValidatorImpl::VerifyRsSignature(const char *pkey,
+bool JwtValidatorImpl::ExtractPubkeyFromJwkEC(const grpc_json *jkey) {
+  if (eck_ != nullptr) {
+    EC_KEY_free(eck_);
+  }
+  eck_ = EC_KEY_new_by_curve_name(NID_X9_62_prime256v1);
+  if (eck_ == nullptr) {
+    gpr_log(GPR_ERROR, "Could not create ec key.");
+    return false;
+  }
+  const char *eck_x = GetStringValue(jkey, "x");
+  const char *eck_y = GetStringValue(jkey, "y");
+  if (eck_x == nullptr || eck_y == nullptr) {
+    gpr_log(GPR_ERROR, "Missing EC public key field.");
+    return false;
+  }
+  BIGNUM *bn_x = BigNumFromBase64String(&exec_ctx_, eck_x);
+  BIGNUM *bn_y = BigNumFromBase64String(&exec_ctx_, eck_y);
+  if (bn_x == nullptr || bn_y == nullptr) {
+    gpr_log(GPR_ERROR, "Could not generate BIGNUM-type x and y fields.");
+    return false;
```

Comment:
I believe you meant  bn_x and bn_y. Yes,  bn_x and bn_y should be freed to avoid memory leak if EC_KEY_set_public_key_affine_coordinates does not succeed.